### PR TITLE
Kill lock unwraps

### PR DESCRIPTION
**Pull request by gavofyork**
Replace with alternative combined function.

This should make the panicker sweep a bit less tedious.

### CODE REVIEW

File: util/src/misc.rs [hunk 1]
```diff
@@ -65,3 +65,28 @@ pub fn version_data() -> Bytes {
...
 	s.out()
 }
+
+/// Object can be locked directly into a `MutexGuard`.
+pub trait Lockable<T> {
+	/// Lock object directly into a `MutexGuard`.
+	fn locked(&self) -> MutexGuard<T>;
+}
+
+impl<T: Sized> Lockable<T> for Mutex<T> {
```
Comment (rphmeier) [inline util/src/misc.rs]: sized bound is implicit

File: util/src/misc.rs [hunk 2]
```diff
@@ -65,3 +65,28 @@ pub fn version_data() -> Bytes {
...
+/// Object can be read or write locked directly into a guard.
+pub trait RwLockable<T> {
+	/// Read-lock object directly into a `ReadGuard`.
+	fn unwrapped_read(&self) -> RwLockReadGuard<T>;
+
+	/// Write-lock object directly into a `WriteGuard`.
+	fn unwrapped_write(&self) -> RwLockWriteGuard<T>;
+}
+
+impl<T: Sized> RwLockable<T> for RwLock<T> {
```
Comment (rphmeier) [inline util/src/misc.rs]: sized bound is implicit

**Comment by rphmeier** [comment]
glad you did this -- it will also make moving to `parking_lot` locks, which don't return `Result`, easier if we choose to do so

**Comment by gavofyork** [comment]
i'd really like to call it something other than `unlocked_write`. but as far as i understand, there's no way i can override the `write` function through the trait?

i hate putting `_`s on the end of functions. any ideas?

**Comment by tomusdrw** [comment]
Maybe `.for_reading()` `.for_writing()` and `.for_rw() / .for_writing()` for `Mutex` ?

**Comment by gavofyork** [comment]
still not nearly as succinct as `read()` or `write()`.

**Comment by gavofyork** [comment]
@rphmeier can we place the trait functions in favour to the basic `impl` functions?

**Pull request merged**